core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -363,6 +371,70 @@ public boolean isWaitForReady() {
   }
 
   /**
+   * Sets the maximum allowed message size acceptable from the remote peer.  If unset, this will
+   * default to the value set on the {@link ManagedChannel}.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withMaxInboundMessageSize(int maxSize) {
+    checkArgument(maxSize >= 0, "invalid maxsize %s", maxSize);
+    CallOptions newOptions = new CallOptions(this);
+    newOptions.maxInboundMessageSize = maxSize;
+    return newOptions;
+  }
+
+  /**
+   * Sets the maximum allowed message size acceptable sent to the remote peer.  If unset, this will
+   * default to the value set on the {@link ManagedChannel}.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withMaxOutboundMessageSize(int maxSize) {
+    checkArgument(maxSize >= 0, "invalid maxsize %s", maxSize);
+    CallOptions newOptions = new CallOptions(this);
+    newOptions.maxOutboundMessageSize = maxSize;
+    return newOptions;
+  }
+
+  /**
+   * Clears the maximum allowed message size acceptable from the remote peer.  The value set on the
+   * {@link ManagedChannel} may still be enforced.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/FIXME")
+  public CallOptions withoutMaxInboundMessageSize() {
```

Comment:
API consistency is also important. I don't want to see two styles of reverting to channel default on the same interface.

On the other hand, I don't think boxing vs primitive in this particular case makes any difference in terms of memory allocation. Suppose we find out the following code allocates so many Integers that we want to get rid of them:
```java
stub.withMaxInboundMessageSize(maxSize).hello(req);
```
It's also allocating that many `CallOptions` at the same time, and we would also have the reason to optimize it out since it's larger than Integer.  The optimization would likely be saving the configured stub instead of configuring it per-call. After we stop allocating that many `CallOptions`, the boxing if Integers will become a non-issue.